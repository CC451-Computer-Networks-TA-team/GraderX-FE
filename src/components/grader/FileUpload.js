import React, { useState } from "react";
import { Button, Label, Header, Icon } from "semantic-ui-react";

export function validateExtension(extension) {
  const supportedTypes = [
    "application/zip",
    "application/vnd.rar",
    "application/x-7z-compressed"
  ];
  return supportedTypes.includes(extension);
}

function FileUpload(props) {
  const [selectedInvalid, setSelectedInvalid] = useState(false);
  const fileInputRef = React.createRef();

  function onFileChange(event) {
    if (
      event.target.files[0] &&
      validateExtension(event.target.files[0].type)
    ) {
      props.setFile(event.target.files[0]);
    } else {
      setSelectedInvalid(true);
    }
  }

  return (
    <React.Fragment>
      <Header as="h4">
        Select a compressed file containing the submissions
      </Header>
      <Button
        content="Choose File"
        labelPosition="right"
        fluid
        basic
        size={"large"}
        icon="upload"
        onClick={() => fileInputRef.current.click()}
      />
      {selectedInvalid ? (
        <Label pointing size={"medium"} basic color="red">
          Supported types: rar, 7z, zip
        </Label>
      ) : null}
      {/* a hidden input is created because semantic ui does not have a file upload component by default
          a normal button is created that triggers a hidden file input when the button is clicked */}
      <input
        ref={fileInputRef}
        name="submissions_file"
        type="file"
        hidden
        onChange={onFileChange}
        accept=".zip,.rar,.7zip,.tar"
      />
      <Header as="h5" textAlign="center">
        <a
          href={() => false}
          style={{ cursor: "pointer" }}
          onClick={props.resetLab}
        >
          <Icon name="chevron left" style={{ marginRight: ".4em" }} />
          Select another lab
        </a>
      </Header>
    </React.Fragment>
  );
}

export default FileUpload;
