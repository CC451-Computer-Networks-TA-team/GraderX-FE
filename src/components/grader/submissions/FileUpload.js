import React, { useState } from "react";
import { Button } from 'carbon-components-react';
import { Upload16 } from '@carbon/icons-react';
import {validateExtension} from '../../../utils'



function FileUpload(props) {
  const [selectedInvalid, setSelectedInvalid] = useState(false);
  const fileInputRef = React.createRef();

  function onFileChange(event) {
    if (
      event.target.files[0] &&
      validateExtension(event.target.files[0].type)
    ) {
      props.setFile(event.target.files[0]);
    } else {
      setSelectedInvalid(true);
    }
  }


  return (
    <React.Fragment>
      <p> Only .rar / .7z / .zip files. 500mb max file size </p>
      <div style={{ height: 16 }}></div>
      <Button renderIcon={Upload16} onClick={() => fileInputRef.current.click()}>Select file</Button>
      {selectedInvalid ? (
        <div>
          <div style={{ height: 16 }}></div>
          <p>Supported files: .rar / .7z / .zip</p>
        </div>
      ) : null}
      {/* a hidden input is created because semantic ui does not have a file upload component by default
            a normal button is created that triggers a hidden file input when the button is clicked */}
      <input
        ref={fileInputRef}
        name="submissions_file"
        type="file"
        hidden
        onChange={onFileChange}
        accept=".zip,.rar,.7zip,.tar"
      />
    </React.Fragment>
  );
}

export default FileUpload;
