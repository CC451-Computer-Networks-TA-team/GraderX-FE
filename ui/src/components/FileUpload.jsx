import React , {useState} from 'react';
import axios from 'axios'; 

function FileUpload(){
    const [selectedFile , selectFile] = useState(null);

    function onFileChange (event){ 
        console.log(event.target);
        selectFile(event.target.files[0])
    }; 

    function onFileUpload(){ 
        const formData = new FormData(); 
        formData.append( 
            "myFile", 
            selectedFile, 
            selectedFile.name 
        ); 
        console.log(selectedFile); 
        axios.post("api/uploadfile", formData); 
    }; 

    function fileData(){ 
        if (selectedFile) { 
            return ( 
                <div> 
                    <h2>File Details:</h2> 
                    <p>File Name: {selectedFile.name}</p> 
                    <p>File Type: {selectedFile.type}</p> 
                    <p> 
                    Last Modified:{" "} 
                    {selectedFile.lastModifiedDate.toDateString()} 
                    </p> 
                </div> 
            ); 
        }else{ 
            return ( 
                <div> 
                    <br /> 
                    <h4>Choose before Pressing the Upload button</h4> 
                </div> 
            ); 
        } 
    }; 

    return ( 
		<div> 
			<h3> 
			Upload a .zip file containing the lab submission. 
			</h3> 
			<div> 
				<input type="file" onChange={onFileChange} /> 
				<button onClick={onFileUpload}> 
				Upload! 
				</button> 
			</div> 
		    {fileData()} 
		</div> 
	); 

}

export default FileUpload; 
