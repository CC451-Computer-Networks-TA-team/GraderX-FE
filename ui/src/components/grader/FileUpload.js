import React, { useState } from "react";
import apiClient from "../../api-client";

function FileUpload(props) {
  const [selectedFile, selectFile] = useState(null);

  function onFileChange(event) {
    selectFile(event.target.files[0]);
  }

  function onFileUpload() {
    const formData = new FormData();
    formData.append("submissions_file", selectedFile);
    apiClient.uploadSubmissions(props.lab, formData).then(res => {
      props.resultsReady(true);
    });
  }

  function fileData() {
    if (selectedFile) {
      return (
        <div>
          <h2>File Details:</h2>
          <p>File Name: {selectedFile.name}</p>
          <p>File Type: {selectedFile.type}</p>
          <p>Last Modified: {selectedFile.lastModifiedDate.toDateString()}</p>
        </div>
      );
    } else {
      return (
        <div>
          <br />
          <h4>Choose before Pressing the Upload button</h4>
        </div>
      );
    }
  }

  return (
    <div>
      <h3>Upload a .rar file containing the lab submissions.</h3>
      <div>
        <input type="file" onChange={onFileChange} />
        {selectedFile ? (
          <button onClick={onFileUpload}>Upload and Grade</button>
        ) : null}
      </div>
      {fileData()}
    </div>
  );
}

export default FileUpload;
