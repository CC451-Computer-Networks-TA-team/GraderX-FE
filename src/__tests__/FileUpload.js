import React, { useState } from "react";
import apiClient from "../api-client";
//import {render, cleanup} from 'react-testing-library'
import FileUpload from '../components/grader/FileUpload';
import renderer from 'react-test-renderer';


test('Dummy Test', () => {
  const component = renderer.create(<FileUpload/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
});