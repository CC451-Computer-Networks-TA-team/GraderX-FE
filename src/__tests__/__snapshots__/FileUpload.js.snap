// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy Test 1`] = `
<div>
  <h3>
    Upload a .rar file containing the lab submissions.
  </h3>
  <div>
    <input
      onChange={[Function]}
      type="file"
    />
  </div>
  <div>
    <br />
    <h4>
      Choose before Pressing the Upload button
    </h4>
  </div>
</div>
`;
